src/server/scripts/Kalimdor/Firelands/boss_ragnaros_firelands.cpp

```diff
@@ -0,0 +1,5304 @@
+            {
+                summons.Summon(summon);
+                if (me->IsInCombat())
+                    DoZoneInCombat(summon, SIZE_OF_GRIDS);
+
+                switch (summon->GetEntry())
+                {
+                    case NPC_SULFURAS_SMASH_TRIGGER:
+                        _inSulfurasSmash = true;
+                        me->CastStop();
+                        me->AttackStop();
+                        me->SetReactState(REACT_PASSIVE);
+                        me->SetFacingToObject(summon);
+                        me->UpdatePosition(me->GetPositionX(), me->GetPositionY(), me->GetPositionZ(), me->GetAngle(summon));
+                        DoCastAOE(SPELL_SULFURAS_SMASH);
+                        DoCastAOE(SPELL_SUMMON_SULFURAS_SMASH, true);
+                        Talk(EMOTE_SULFURAS_SMASH);
+                        events.DelayEvents(6 * IN_MILLISECONDS, EVENT_GROUP_DELAYABLE);
+                        events.ScheduleEvent(EVENT_RESUME_ATTACK, 5.6 * IN_MILLISECONDS);
+                        break;
+                    case NPC_LAVA_SCION:
+                        if (instance)
+                            instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, summon);
+                        break;
+                    case NPC_CENARIUS:
+                    case NPC_HAMUUL_RUNETOTEM:
+                    case NPC_MALFURION_STORMRAGE:
+                        summon->setActive(true);
+                        instance->instance->LoadGrid(summon->GetPositionX(), summon->GetPositionY());
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+            void SummonedCreatureDies(Creature* summon, Unit* killer) override
+            {
+                switch (summon->GetEntry())
+                {
+                    case NPC_SON_OF_FLAME:
+                        summons.Despawn(summon);
+                        if (events.IsInPhase(PHASE_INTERMISSION) && !summons.HasEntry(NPC_SON_OF_FLAME))
+                            PreparePhase(_previousPhase == PHASE_ONE ? PHASE_TWO : PHASE_THREE);
+                        break;
+                    case NPC_LAVA_SCION:
+                        if (instance)
+                            instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, summon);
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+            void SummonedCreatureDespawn(Creature* summon) override
+            {
+                switch (summon->GetEntry())
+                {
+                    case NPC_LAVA_SCION:
+                        if (instance)
+                            instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, summon);
+                        break;
+                    default:
+                        break;
+                }
+
+                if (summon->IsAIEnabled)
+                    summon->AI()->EnterEvadeMode();
```

Comment:
Hm...
Can't remember why I wrote this.
Might be because I've seen creatures in encounters returning to their spawn point prior to being despawned, when wiping on a boss, so I just threw this in here to mimic that same behavior.
Probably something we can skip.